src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -187,87 +182,81 @@ public void Dispose()
         }
 
         /// <summary>
-        /// This function will evaluate given string and it will try to convert
-        /// it to a Uri object. The string could contain URI fragment, IP address and  port
-        /// tuple or just IP address or name. It will return null if parsing fails.
+        /// This function is used to parse WinINet Proxy strings. The strings are a semicolon
+        /// or whitespace separated list, with each entry in the following format:
+        /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
         /// </summary>
-        private static Uri GetUriFromString(string value)
+        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy)
         {
+            insecureProxy = null;
+            secureProxy = null;
             if (string.IsNullOrEmpty(value))
             {
-                return null;
+                return;
             }
 
-            if (!value.Contains("://"))
+            if (value.IndexOfAny(s_proxyDelimiters) == -1)
```

Comment:
In terms of performance, I'm not sure if 'IndexOfAny' is the best choice.  But @stephentoub could comment on what the lastest/fastest way to this string is (I know that .NET Core 3.0 has improved various ways of doing it).

Reply:
There are some changes I can make to reduce allocations if we're aiming for max perf, though I don't know if this method is significant enough to warrant it.

Reply:
This code gets executed only once for the whole process, right?  Assuming yes, I'd opt for simple-and-not-terrible over complicated-and-super-fast.